src/encoded/static/components/regulome_search.js

```diff
@@ -544,6 +544,32 @@ const populationOrder = [
     'source unknown',
 ];
 
+// function to de-dup overlapping peaks in each dataset from ChIP, DNase and ATAC-seq assays
+function filter_overlapping_peaks(Datasets) {
+    var Datasets_filtered = [];
+    var Datasets_filtered_count = 0;
+    if (Datasets.length > 0) {
+        //we want to keep the wider peaks in each dataset
+        //if the start & end positions are the same we will keep the one with the strongest signal
+        //sort all peaks by dataset ids -> start positions in ascending order -> end positions in descending order -> signals in descending order
+        Datasets.sort((a,b)=>a.dataset_rel.localeCompare(b.dataset_rel) || a.start-b.start || b.end-a.end || b.value-a.value);
+        //keep the peak if it is the first peak in a new dataset or if it is not within the previous peak
+        Datasets_filtered.push(Datasets[0]);
+        var lastDataset = Datasets[0].dataset_rel;
+        var lastEnd =  Datasets[0].end;
+        for (var i = 1; i < Datasets.length; i++) {
+            if (Datasets[i].dataset_rel !== lastDataset || Datasets[i].end > lastEnd) { 
+                Datasets_filtered.push(Datasets[i]); 
+                lastDataset = Datasets[i].dataset_rel;
+                lastEnd = Datasets[i].end;
+            } else {
+                Datasets_filtered_count ++;
+            }
+        }
+    }
+    return [Datasets_filtered,Datasets_filtered_count];
```

Comment:
Should have one space after comma. Did you lint your code before pushing?

Reply:
Thanks for pointing out that, my eslint wasn't working correctly. I have also fixed other stylistic issues in my code.